src/components/app/content/card/index.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import './index.css';
+import { updateCard } from '../../../../redux/actions';
+import { connect } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+function Card(props) {
+    const [tempState, setTempState] = useState(props.location.cardData);
+    const history = useHistory();
+
+    const fillData = (data) => setTempState({ ...tempState, ...data });
+
+    const saveCardHandler = () => {
+        props.updateCard(tempState);
+        history.push('/');
+    };
+
+    return (
+        <div className="container">
+            <div className="single-card">
+                <div className="single-card-header">
+                    <input
+                        type="text"
+                        value={tempState.header}
+                        onChange={(event) =>
+                            fillData({ header: event.target.value })
+                        }
+                    />
+                    <button
+                        className="btn btn-primary"
+                        onClick={() => saveCardHandler()}
+                    >
+                        Save card changes
+                    </button>
+                </div>
+                <div className="single-card-body">
+                    <textarea
+                        value={tempState.body}
+                        onChange={(event) => fillData({ body: event.target.value })}
+                    />
+                </div>
+            </div>
+        </div>
+    );
```

Comment:
You already have Card component - it displays cards in the card list. Why not just re-use it? React is all about reusing the  components... 
Also, make sure single card component provides the same functionality as cards in the list - both edit and view mode with corresponding icons.